### PR TITLE
[Firebase_core] Migrate to AndroidX

### PR DESCRIPTION
**Pull request by ashutoshkmr**
## Description

Migrate Firebase_core to AndroidX

## Related Issues

flutter/flutter#40540

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_messaging/example/android/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.3.0'
+        classpath 'com.android.tools.build:gradle:3.5.0'
```
Comment (collinjackson) [inline packages/firebase_messaging/example/android/build.gradle]: did you mean to modify this file?

File: packages/firebase_messaging/example/android/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,6 @@
+#Mon Sep 16 16:11:29 IST 2019
```
Comment (collinjackson) [inline packages/firebase_messaging/example/android/gradle/wrapper/gradle-wrapper.properties]: this seems unrelated to the firebase_core change

File: packages/firebase_core/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## 0.5.0
+
+* Migrate To AndroidX.
+* This is a breaking change and requires manual migration. [Click here for more details](https://flutter.dev/docs/development/packages-and-plugins/androidx-compatibility#how-to-migrate-a-flutter-app-to-androidx)
```
Comment (collinjackson) [inline packages/firebase_core/CHANGELOG.md]: Can you clarify what situations are being improved by this change? I don't think we want to force Flutterfire users to manually migrate their apps.

I'm wondering if landing https://github.com/FirebaseExtended/flutterfire/pull/39 would be sufficient for your use case.

**Comment by ashutoshkmr** [comment]
my issue has been resolved, thanks!!
